src/common/src/config.rs

```diff
@@ -366,12 +366,16 @@ pub struct ServerConfig {
     #[serde(default = "default::server::telemetry_enabled")]
     pub telemetry_enabled: bool,
 
-    #[serde(default, flatten)]
-    pub unrecognized: Unrecognized<Self>,
-
     /// Enable heap profile dump when memory usage is high.
     #[serde(default)]
     pub auto_dump_heap_profile: AutoDumpHeapProfileConfig,
+
+    /// Manually heap profile dump file path.
+    #[serde(default = "default::server::manually_dump_heap_profile_dir")]
+    pub manually_dump_heap_profile_dir: String,
```

Comment:
Is it necessary to specify a directory? In my opinion using a `tempfile` or `tempdir` should be enough.

Reply:
There should be a directory to store dumped file users wants to analyze later

Reply:
Yeah, but IMO just dump into a temporary directory and log the path, users can find them out as well. 🤣 `/tmp` is durable enough here.

Reply:
Considering OOM will cause pod restarts, `/tmp` is not persisted. Actually all folders are not persisted except the mounted folders (aka. [volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)). Thus we need to use the user provided folder.


However, I do think it's bothersome to pass two directories here. Recommend using one configurable option for both auto dump and manual dump.